src/bionics.cpp

```diff
@@ -67,8 +69,125 @@ const efftype_id effect_visuals( "visuals" );
+            pos[0] = 2;
+            pos[1] = std::max( getmaxx( win ) / 2 - utf8_width( titles[1] ) / 2 ,
+                               pos[0] + utf8_width( titles[0] ) + 2 );
+            pos[2] = std::max( getmaxx( win ) * 2 / 3 + 7,
+                               pos[1] + utf8_width( titles[1] ) + 2);
+
+        for( size_t i = 0; i < std::min( titles.size(), pos.size() ); ++i ) {
+            mvwprintw( win, 3, pos[i], titles[i].c_str() );
+        }
+    }
+    wrefresh( win );
+}
+
+void draw_header( WINDOW *win, const size_t tab_index, const std::string power_string,
+                  const std::string help_key )
+{
+    const std::array<std::string, 3> titles = {{
+        string_format( _( "Body Parts <color_yellow><< %s >></color>" ), bodyparts[tab_index].c_str() ),
+        power_string,
+        string_format( _( "Press '<color_yellow>%s</color>' for help" ), help_key.c_str() )
+    }};
+    // left / center / right aligned elements
+    const std::array<int, 3> pos = {{
+        0,
+        ( getmaxx( win ) - utf8_width( titles[1] ) ) / 2 + 1,
+        ( getmaxx( win ) - utf8_width( titles[2], true ) - 1 )
+    }};
+
+    werase( win );
+    for( size_t i = 0; i < std::min( titles.size(), pos.size() ); ++i ) {
+        fold_and_print( win, 0, pos[i], getmaxx( win ) - 3, c_white, titles[i] );
+    }
+    wrefresh( win );
+}
+
+const auto separator = []( std::ostringstream &s )
+{
+    return s.tellp() != 0 ? ", " : "";
+};
+
+std::string power_description( std::string const &id )
+{
+    std::ostringstream power_desc;
+    if( bionics[id].power_over_time > 0 && bionics[id].charge_time > 0 ) {
+        power_desc << ( bionics[id].charge_time == 1 ?
+                        string_format( _("%d PU / turn" ), bionics[id].power_over_time ) :
+                        string_format( _("%d PU / %d turns" ), bionics[id].power_over_time,
+                                       bionics[id].charge_time ) );
+    }
+    if( bionics[id].power_activate > 0 && bionics[id].charge_time == 0 ) {
+        power_desc << separator( power_desc ) << string_format( _( "%d PU / activation" ),
+                                                                bionics[id].power_activate );
+    }
+    if( bionics[id].power_deactivate > 0 && bionics[id].charge_time == 0 ) {
+        power_desc << separator( power_desc ) << string_format( _( "%d PU / deactivation" ),
+                                                                bionics[id].power_deactivate );
+    }
+    return power_desc.str();
+}
+
+std::vector<cbm_pair> define_content( player& u, body_part bp )
+{
+    std::vector<cbm_pair> content;
+    content.push_back( std::make_pair( bp , INT_MAX ) );
+    for( size_t i = 0; i < u.my_bionics.size(); ++i ) {
+        if( g->u.my_bionics[i].occupied_bp == bp ) {
+            content.push_back( std::make_pair( bp, i ) );
+        }
+    }
+    std::sort( content.begin(), content.end(), [&u] ( cbm_pair const &val1, cbm_pair const &val2 ) {
```

Comment:
this is the very first time I used lambda so it would be good to receive confirmation I did everything right.